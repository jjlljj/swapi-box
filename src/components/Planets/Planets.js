import React, { Component } from 'react';
import propTypes from 'prop-types';
import './Planets.css';
import Card from '../Card/Card';

class Planets extends Component {
   
  async componentDidMount() {
    if (!this.props.cards) this.props.fetchData()
  }

  renderCards() {
    return this.props.cards.map(card => {
            return (<Card card={card} />)
    })
  }

  render() {
    return (
      <div className="card-container"> 
        { 
          this.props.cards &&
          this.renderCards()
        }
      </div>
    );
  }
};

export default Planets;
