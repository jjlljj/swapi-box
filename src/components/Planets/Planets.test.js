/* eslint-disable */

import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Planets from './Planets';

describe('Planets', () => {

  it('should not pass', () => {
    

    expect(true).toEqual(false)
  })

})
