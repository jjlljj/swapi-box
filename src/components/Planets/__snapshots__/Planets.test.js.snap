// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planets should pass snapshot when passed a single card 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Planets
    addToFav={undefined}
    cards={
        Array [
            Object {
              "Climate": "temperate, tropical",
              "Population": "1000",
              "Residents": "",
              "Terrain": "jungle, rainforests",
              "cardType": "planets",
              "favorite": true,
              "name": "Yavin IV",
            },
          ]
    }
    fetchData={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          addToFav={undefined}
          card={
                    Object {
                              "Climate": "temperate, tropical",
                              "Population": "1000",
                              "Residents": "",
                              "Terrain": "jungle, rainforests",
                              "cardType": "planets",
                              "favorite": true,
                              "name": "Yavin IV",
                            }
          }
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Yavin IV",
        "nodeType": "function",
        "props": Object {
          "addToFav": undefined,
          "card": Object {
            "Climate": "temperate, tropical",
            "Population": "1000",
            "Residents": "",
            "Terrain": "jungle, rainforests",
            "cardType": "planets",
            "favorite": true,
            "name": "Yavin IV",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            addToFav={undefined}
            card={
                        Object {
                                    "Climate": "temperate, tropical",
                                    "Population": "1000",
                                    "Residents": "",
                                    "Terrain": "jungle, rainforests",
                                    "cardType": "planets",
                                    "favorite": true,
                                    "name": "Yavin IV",
                                  }
            }
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Yavin IV",
          "nodeType": "function",
          "props": Object {
            "addToFav": undefined,
            "card": Object {
              "Climate": "temperate, tropical",
              "Population": "1000",
              "Residents": "",
              "Terrain": "jungle, rainforests",
              "cardType": "planets",
              "favorite": true,
              "name": "Yavin IV",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Planets should pass snapshot when passed cards 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Planets
    addToFav={undefined}
    cards={
        Array [
            Object {
              "Climate": "murky",
              "Population": "unknown",
              "Residents": "",
              "Terrain": "swamp, jungles",
              "cardType": "planets",
              "favorite": true,
              "name": "Dagobah",
            },
            Object {
              "Climate": "temperate, tropical",
              "Population": "1000",
              "Residents": "",
              "Terrain": "jungle, rainforests",
              "cardType": "planets",
              "favorite": true,
              "name": "Yavin IV",
            },
          ]
    }
    fetchData={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          addToFav={undefined}
          card={
                    Object {
                              "Climate": "murky",
                              "Population": "unknown",
                              "Residents": "",
                              "Terrain": "swamp, jungles",
                              "cardType": "planets",
                              "favorite": true,
                              "name": "Dagobah",
                            }
          }
/>,
        <Card
          addToFav={undefined}
          card={
                    Object {
                              "Climate": "temperate, tropical",
                              "Population": "1000",
                              "Residents": "",
                              "Terrain": "jungle, rainforests",
                              "cardType": "planets",
                              "favorite": true,
                              "name": "Yavin IV",
                            }
          }
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Dagobah",
        "nodeType": "function",
        "props": Object {
          "addToFav": undefined,
          "card": Object {
            "Climate": "murky",
            "Population": "unknown",
            "Residents": "",
            "Terrain": "swamp, jungles",
            "cardType": "planets",
            "favorite": true,
            "name": "Dagobah",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Yavin IV",
        "nodeType": "function",
        "props": Object {
          "addToFav": undefined,
          "card": Object {
            "Climate": "temperate, tropical",
            "Population": "1000",
            "Residents": "",
            "Terrain": "jungle, rainforests",
            "cardType": "planets",
            "favorite": true,
            "name": "Yavin IV",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            addToFav={undefined}
            card={
                        Object {
                                    "Climate": "murky",
                                    "Population": "unknown",
                                    "Residents": "",
                                    "Terrain": "swamp, jungles",
                                    "cardType": "planets",
                                    "favorite": true,
                                    "name": "Dagobah",
                                  }
            }
/>,
          <Card
            addToFav={undefined}
            card={
                        Object {
                                    "Climate": "temperate, tropical",
                                    "Population": "1000",
                                    "Residents": "",
                                    "Terrain": "jungle, rainforests",
                                    "cardType": "planets",
                                    "favorite": true,
                                    "name": "Yavin IV",
                                  }
            }
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Dagobah",
          "nodeType": "function",
          "props": Object {
            "addToFav": undefined,
            "card": Object {
              "Climate": "murky",
              "Population": "unknown",
              "Residents": "",
              "Terrain": "swamp, jungles",
              "cardType": "planets",
              "favorite": true,
              "name": "Dagobah",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Yavin IV",
          "nodeType": "function",
          "props": Object {
            "addToFav": undefined,
            "card": Object {
              "Climate": "temperate, tropical",
              "Population": "1000",
              "Residents": "",
              "Terrain": "jungle, rainforests",
              "cardType": "planets",
              "favorite": true,
              "name": "Yavin IV",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
